docs/DeveloperGuide.md

```diff
@@ -1,34 +1,851 @@
+
+1. `printBreakdown()` then calculates the total time spent by the user on the modules taken in the specified 
+week. It then prints out `actualTime` of each `Module` as a percentage of `totalTimeSpent`. If there is no input, 
+a `NO INPUT` message will be printed. `printBreakdown()` return **TRUE** if there exists a `Module` that has a 
+valid input, else **False** is returned.
+
+1. If output of `printBreakdown()` is `True`, `printAnalysis()` will be called. `computeAnalysisOfTimeSpent()` 
+is called to determine the outcome of the analysis. A message will be printed depending on the output
+of the analysis. 
+
+The following sequence diagram shows how the analysis command works. 
+
+![analysis](diagrams/AnalysisSequence.png)
+
+#### Design Considerations
+
+* **Alternative 1 (current choice)**: Printing data as a bar graph
+    * Pros: Easy implementation as the bar graph is printed as a `String`.
+    * Cons: It might be difficult to compare across modules using a bar graph.
+* **Alternative 2**: Printing data as a line graph
+    * Pros: Easier to compare across different modules.
+    * Cons: Difficult to implement as it requires external libraries.
+    * Cons: Difficult to test due to additional dependencies on external libraries. 
+
+### 4.7 Add Task <a name="4.7"></a>
+The add task feature allows user to add a task under an existing module. 
+
+#### Current Implementation
+The `addtask` command is executed by the `parse` method in the `Parser` class, which then calls the `addTask` method 
+in the `TaskList` class. The `addTask` method performs all the necessary logic for the add task feature.
+
+Given below is an example usage scenario.
+1. The user input the `addtask` command and the `parse` method in `Parser` parses the command.
+1. `parse` calls the `addTask` method in `TaskList`.
+1. The `addTask` method splits the user input into 3 sections:
+    * `Section 1`: command
+    * `Section 2`: module code
+    * `Section 3`: task description
+1. The `addTask` method calls the `checkIfModuleValid` method in `ModuleList` class to check if `Section 2` is a 
+valid module code.
+    * If module code is not valid, the `addTask` method will return and `Ui` will prompt the user to enter a 
+    valid module code.
+    * Else if module code is valid, the `addTask` method will execute the next step.
+1. The `addTask` method calls the `checkIfModuleExist` method to check if `Secion 2` (module code) exists in the database.
+    * If module code does not exist in the database, the `addTask` method will return and `Ui` will prompt the user to
+    execute the `addmod` or `addexp` command first.
+    * Else if module code exists, the `addTask` method will execute the next step.
+1. The `addTask` method will create a new `Task` object by the constructor in `Task` class, with `Section 3` (task description). 
+1. The `addTask` method adds the `Task` object to the array list `tasks`.
+
+![addtask](diagrams/addtask.png)
```

Comment:
![image](https://user-images.githubusercontent.com/60372589/97388757-b4017700-1913-11eb-94cd-eab391e703ef.png)

Suggestion: Should there are a return dotted arrow

Reply:
Should :TaskList return control to :Parser instead of User since it was called by :Parser? Also, from earlier sections, I thought the Parser is called by :Ui instead of directly by User?

![97535285-61e55200-19f6-11eb-85f5-b1c1c95c9bfe](https://user-images.githubusercontent.com/48957360/97775744-11a8f400-1b9e-11eb-8413-46ff67dbd360.png)